data/mods/Magiclysm/items/enchanted_melee.json

```diff
@@ -854,6 +854,39 @@
     "use_action": [ "WEATHER_TOOL" ],
     "flags": [ "DURABLE_MELEE", "SHEATH_SWORD", "TRADER_AVOID", "MAGIC_FOCUS", "THERMOMETER" ]
   },
+  {
+    "id": "rune_kelvinist_ice_adept_weapon",
+    "type": "GENERIC",
+    "symbol": "/",
+    "color": "light_gray",
+    "name": { "str": "Kelvinist Frostbind" },
+    "//": "Stats based on scimitar +2, a weapon that will cast freezing touch when it hits, has wide strike",
+    "description": "This curved sword reminiscent of a scimitar has a handle with a crystal attracted to it, you feel a cold wind emanating from it.  There are kelvinist runes engraved in the blade.",
+    "price": 558000,
+    "price_postapoc": 21000,
+    "material": [ "steel" ],
+    "techniques": [ "WBLOCK_2", "WIDE" ],
+    "weight": "960 g",
+    "volume": "2 L",
+    "longest_side": "90 cm",
+    "bashing": 8,
+    "cutting": 36,
+    "to_hit": 3,
+    "category": "weapons",
+    "qualities": [ [ "CUT", 1 ], [ "BUTCHER", 8 ] ],
+    "use_action": [
+      { "type": "cast_spell", "spell_id": "frost_spray", "no_fail": true, "level": 10, "need_wielding": true },
+      "WEATHER_TOOL"
+    ],
+    "flags": [ "DURABLE_MELEE", "SHEATH_SWORD", "TRADER_AVOID", "THERMOMETER", "MAGIC_FOCUS", "NO_UNLOAD", "NO_RELOAD" ],
+    "charges_per_use": 1,
```

Comment:
Looks like you need to swap `"ammo": [ "crystallized_mana" ],` and `"charges_per_use": 1,`

Reply:
I need to put `"ammo": [ "crystallized_mana" ],` where `"charges_per_use":1` is?

Reply:
The problem was that it needed `"type": "TOOL"` for the charges to work.